src/Pos/Block/Worker.hs

```diff
@@ -115,18 +115,20 @@ blkOnNewSlotImpl (slotId@SlotId {..}) sendActions = do
                       map (bprint pairF) (zip [0 :: Int ..] $ toList leaders)
         logLeadersF $ sformat ("Current slot leader: "%build) leader
         logDebugS $ sformat ("Available to use lightweight PSKs: "%listJson) validCerts
-        heavyPskM <- getPSKByIssuerAddressHash leader
-        logDebug $ "Does someone have cert for this slot: " <> show (isJust heavyPskM)
-        let heavyWeAreDelegate = maybe False ((== ourPk) . pskDelegatePk) heavyPskM
-        let heavyWeAreIssuer = maybe False ((== ourPk) . pskIssuerPk) heavyPskM
+        heavyPskIssuerM <- getPskByIssuer (Right leader)
+        let heavyWeAreIssuer = maybe False ((== ourPk) . pskIssuerPk) heavyPskIssuerM
```

Comment:
*OLD COMMENT:*

I don't understand this check. I think we should check whether **we** delegated to somebody?!

Currently you check whether leader delegated to us. But I don't get it. For example, suppose we are leader and we delegated to somebody (not to us, obviously). Then `heavyWeAreIssuer` will be `False`. But in this case we shouldn't create a block, because we delegated to somebody.

Another example. We are not a leader, but leader delegated to us. `heavyWeAreIssuer` will be `True`. And we won't create a block. But we should do it unless we delegated further!

*UPDATE*:

I read it again more carefully. Again, I don't understand what's going on. We get a cert issued by leader and then compare `pskIssuerPk` with `ourPk`. But `pskIssuerPk` must be leader's PK. Am I saying nonsense?